spec/lumberjack_spec.rb

```diff
@@ -76,10 +76,12 @@
     # TODO(sissel): Make sure this doesn't take forever, do a timeout.
     count = 0
     events = []
-    connection = server.accept
-    connection.run do |event|
+    server.run do |event|
       events << event
-      connection.close if events.length == lines.length
+      if events.length == lines.length
+        server.stop
+        raise EOFError.new("Force the connection out of the read loop")
```

Comment:
server.accept seemed like it was reaching in to the internals a bit too far. refactoring the code broke it, and changing the test to use the new accept ignored parts of the code that could have been tested otherwise.